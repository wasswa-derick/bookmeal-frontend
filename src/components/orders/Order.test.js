import React from "react";
import { mount } from "enzyme";
import { Order } from "./Order";

const fn = data => Promise.resolve(data);

describe("Order", () => {
  let wrapper;
  let mockFn;
  beforeEach(() => {
    mockFn = jest.fn();
    const history = {
      push: mockFn
    };
    const match = {
      params: {
        id: "1"
      }
    };

    const menu = {
      id: 1,
      title: "",
      description: "",
      menuDate: "",
      meals: [{ id: 1, title: "", description: "", price: 1000 }]
    };
    wrapper = mount(
      <Order
        menu={menu}
        history={history}
        setMessage={mockFn}
        postOrder={fn}
        getMenu={fn}
        match={match}
      />
    );
  });

  it("should handle cancelOrder and makeOrder", () => {
    wrapper.instance().cancelOrder();
    wrapper.instance().makeOrder();
    expect(mockFn.mock.calls.length).toEqual(1);
  });

  it("should increment or decrement count", () => {
    const evt = {};
    const { order } = wrapper.instance().state;
    expect(order.orderCount).toBe(1);

    // increment count by 1
    wrapper.instance().incrementCount(evt, true);

    const { orderCount } = wrapper.instance().state.order;
    expect(orderCount).toBe(2);

    // then decrement the count by 1
    wrapper.instance().incrementCount(evt, false);
    expect(wrapper.instance().state.order.orderCount).toBe(1);
  });

  it("should add or remove meal to  orders when checked is called", () => {
    const evt = {
      target: { checked: true }
    };
    expect(wrapper.instance().state.order.meals.length).toEqual(0);

    // check the meal to add it to order
    wrapper.instance().checked(evt, 1);
    expect(wrapper.instance().state.order.meals.length).toEqual(1);

    // un check the meal to remove it from order
    evt.target.checked = false;
    wrapper.instance().checked(evt, 1);
    expect(wrapper.instance().state.order.meals.length).toEqual(0);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
