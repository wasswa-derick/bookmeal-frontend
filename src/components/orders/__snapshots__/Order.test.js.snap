// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order should render correctly 1`] = `
<Order
  getMenu={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  match={
    Object {
      "params": Object {
        "id": "1",
      },
    }
  }
  menu={
    Object {
      "description": "",
      "id": 1,
      "meals": Array [
        Object {
          "description": "",
          "id": 1,
          "price": 1000,
          "title": "",
        },
      ],
      "menuDate": "",
      "title": "",
    }
  }
  postOrder={[Function]}
  setMessage={[Function]}
>
  <div
    className="container"
  >
    <h5 />
    <hr />
    <Component
      component="div"
      loaded={true}
      loadedClassName="loadedContent"
      parentClassName="loader"
    >
      <div
        className="loadedContent"
        key="content"
      >
        <div
          className="modal-body"
        >
          <div
            className="form-group meal-order-item"
            key="1"
          >
            <input
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ml-2"
              htmlFor="title"
            />
            <label
              className="float-right"
              htmlFor="price"
              style={
                Object {
                  "color": "#00a1e0",
                }
              }
            >
              UGX
              1000
            </label>
            <p
              style={
                Object {
                  "fontSize": "14px",
                }
              }
            />
          </div>
          <hr />
          <div
            className="row"
          >
            <div
              className="col-md-2 input-group"
            >
              <div
                className="input-group-prepend"
              >
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-minus"
                  />
                </button>
              </div>
              <input
                className="form-control"
                readOnly={true}
                type="text"
                value={1}
              />
              <div
                className="input-group-prepend"
              >
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-plus"
                  />
                </button>
              </div>
            </div>
            <div
              className="col-md-4"
              style={
                Object {
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              UGX 
              0
            </div>
            <div
              className="col-md-6 pull-right"
            >
              <button
                className="btn-save btn btn-primary"
                disabled={true}
                onClick={[Function]}
                type="button"
              >
                Order
              </button>
              <button
                className="btn btn-danger ml-4"
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
            </div>
          </div>
        </div>
      </div>
    </Component>
  </div>
</Order>
`;
